script/clang-format.sh

```diff
@@ -1,6 +1,6 @@
-#!/bin/bash
+#!/bin/sh
```

Comment:
nit: Other scripts use `bash`. Should we unify which shell we use? Inside Google, `bash` is chosen for all shell scripts :)

Reply:
this script requires bash.

Reply:
I am fine with bash. I'll go ahead and make the updates, including to openthread/openthread.